process.rb

```diff
@@ -46,118 +87,149 @@ def build_file(filename, &block)
     end
   end
-end
 
-OfficeElection.find_each do |office_election|
-  build_file("/office_election/#{office_election.id}") do |f|
-    f.puts JSON.pretty_generate(office_election.as_json.merge(ballot_id: 1))
+  # /_data/candidates/oakland/2016-11-06/libby-schaaf.json
+  OaklandCandidate
+    .where(election_name: election_name)
+    .includes(:office_election, :calculations)
+    .find_each do |candidate|
+      filename = slugify(candidate['Candidate'])
+      build_file("/_data/candidates/#{locality}/#{election[:date]}/#{filename}.json") do |f|
+        f.puts candidate.to_json
+      end
   end
-end
 
-OaklandCandidate.includes(:office_election, :calculations).find_each do |candidate|
-  %W[
-    /candidate/#{candidate.id}
-  ].each do |candidate_filename|
-    build_file(candidate_filename) do |f|
-      #
-      # To add a field to one of these endpoints, add it to the '#as_json'
-      # method in models/oakland_candidate.rb
-      #
-      f.puts candidate.to_json
+
+  # /_office_elections/oakland/2018-11-06/city-auditor.md
+  office_elections.each do |office|
+    build_file("/_office_elections/#{locality}/#{election[:date]}/#{slugify(office)}.md") do |f|
+      candidates = OaklandCandidate.where(Office: office, election_name: election_name).pluck(:Candidate)
+
+      f.puts(YAML.dump(
+        'title' => office,
+        'candidates' => candidates.map { |name| slugify(name) },
+        'ballot' => ballot_name[1..-1],
+      ))
+      f.puts('---')
     end
   end
 end
 
+# /_committees/1386416.md
+OaklandCommittee.find_each do |committee|
+  build_file("/_committees/#{committee.Filer_ID}.md") do |f|
+    f.puts(YAML.dump(
+      'filer_id' => committee.Filer_ID,
+      'name' => committee.Filer_NamL,
+    ))
+    f.puts('---')
+  end
+end
+
+# /_data/contributions/1229791.json
 OaklandCommittee.includes(:calculations).find_each do |committee|
   next if committee['Filer_ID'].nil?
   next if committee['Filer_ID'] =~ /pending/i
 
-  build_file("/committee/#{committee['Filer_ID']}") do |f|
-    f.puts committee.to_json
+  build_file("/_data/committees/#{committee['Filer_ID']}.json") do |f|
+    f.puts JSON.pretty_generate(contributions: committee.calculation(:contribution_list) || [])
   end
+end
 
-  build_file("/committee/#{committee['Filer_ID']}/contributions") do |f|
-    f.puts JSON.pretty_generate(committee.calculation(:contribution_list) || [])
-  end
-  build_file("/committee/#{committee['Filer_ID']}/opposing") do |f|
-    f.puts JSON.pretty_generate(committee.calculation(:opposition_list) || [])
+OaklandReferendum.includes(:calculations).find_each do |referendum|
+  locality, _year = referendum.election_name.split('-', 2)
+  election = ELECTIONS[referendum.election_name]
+  title = slugify(referendum['Short_Title'])
+
+  if election.nil?
+    $stderr.puts "MISSING ELECTION:"
+    $stderr.puts "  Election Name: #{referendum.election_name}"
+    $stderr.puts '  Add it to ELECTIONS global in process.rb'
+    next
```

Comment:
Maybe you should just throw here? I feel like we want to break the build when something is wrong, rather than publish incomplete data.